### PR TITLE
Add Rose optimizer

### PR DESCRIPTION
**Pull request by MatthewK78**
Rose optimizer: https://github.com/MatthewK78/Rose

- Stateless: uses less memory than even 8-bit versions of other optimizers. If not for temporary working memory, its memory usage would be as low as plain vanilla SGD (**without** momentum).
- Matches or exceeds AdamW in observed accuracy and convergence.
- Training loss runs higher while validation loss runs lower, consistent with implicit regularization rather than memorization.

Note: Requires a higher than usual learning rate. For example, 1e-3 instead of AdamW's 1e-4.

### CODE REVIEW

**Comment by funwithforks** [comment]
This optimizer is quite good.
